### PR TITLE
Feature: Promoted realestate cell

### PR DESCRIPTION
**Pull request by bstien**
# Why?
Realestate has a new promotion/highlight product that needs a new cell within the search results.

The product customers can buy is either pluss or premium highlight. Both variants will be a much larger cell than what we have today.

**Pluss**: A large image.
**Premium**: Two large images + map pinpointing the property.

# What?
- Added `PromotedRealestateCellView`, view model and subviews.

# Version Change

Minor.

# UI Changes

| Pluss | Premium |
| --- | --- |
| Screenshot 2021-06-03 at 10 39 26 | Screenshot 2021-06-03 at 10 39 33 |

### CODE REVIEW

File: Sources/Recycling/PromotedCells/PromotedRealestateCellView.swift [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+    @objc func handleFavoriteButtonTap(_ button: UIButton) {
+        delegate?.promotedRealestateCellViewDidToggleFavoriteState(self, button: button)
+    }
+}
+
+// MARK: - Private extension
+
+private extension UILabel {
+    func setTextOrHideIfEmpty(_ text: String?) {
```
Comment (ag12) [inline Sources/Recycling/PromotedCells/PromotedRealestateCellView.swift:193]: 👌

**Review by ag12 — APPROVED** [review]
Great stuff 🔥

File: Sources/Recycling/PromotedCells/Subviews/FavoriteButton.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import UIKit
+import FinniversKit
+
+class FavoriteButton: UIButton {
```
Comment (hakloev) [inline Sources/Recycling/PromotedCells/Subviews/FavoriteButton.swift:4]: Is there any difference between this and `IconButton` with the `.favorite` style?

Comment (bstien) [reply Sources/Recycling/PromotedCells/Subviews/FavoriteButton.swift:4]: No difference, except the fact that I didn't know about this one😄 Thanks! Will replace and use this instead.

**Pull request merged**